jerry-core/parser/js/js-lexer.c

```diff
@@ -1429,11 +1429,36 @@ lexer_parse_number (parser_context_t *context_p) /**< context */
       }
 
       lexer_check_numbers (context_p, &source_p, source_end_p, LIT_CHAR_7, true);
+
+      if (source_p[0] >= LIT_CHAR_8
+          && source_p[0] <= LIT_CHAR_9)
+      {
+#if ENABLED (JERRY_ESNEXT)
+        lexer_check_numbers (context_p, &source_p, source_end_p, LIT_CHAR_9, false);
+        context_p->token.extra_value = LEXER_NUMBER_DECIMAL;
+#else
+        parser_raise_error (context_p, PARSER_ERR_INVALID_NUMBER);
+#endif /* ENABLED (JERRY_ESNEXT) */
+      }
     }
     else if (source_p[1] >= LIT_CHAR_8
              && source_p[1] <= LIT_CHAR_9)
     {
+      if (context_p->status_flags & PARSER_IS_STRICT)
+      {
+        parser_raise_error (context_p, PARSER_ERR_OCTAL_NUMBER_NOT_ALLOWED);
+      }
+#if ENABLED (JERRY_ESNEXT)
+      lexer_check_numbers (context_p, &source_p, source_end_p, LIT_CHAR_9, true);
```

Comment:
Why `is_legacy` is true?

Reply:
The following test-cases need to throw a syntax error: 08_0.